File: masar_assets/masar_assets/report/users_summary/users_summary.js
// Copyright (c) 2024, KCSC and contributors
// For license information, please see license.txt

frappe.query_reports["Users Summary"] = {
	"filters": [

		{
			"fieldname" : "username",
			"label" : __("Username"),
			"fieldtype" : "Data",
		},
		{
			"fieldname" : "role_profile_name",
			"label" : __("Role Profile"),
			"fieldtype" : "Link",
			"options" : "Role Profile"
		},
        {
            "fieldname": "enabled",
            "label": __("Enabled"),
            "fieldtype": "Select",
            "options": " \nEnabled\nDisabled"
        }
	]
};

File: masar_assets/masar_assets/report/users_summary/users_summary.py
import frappe

def execute(filters=None):
    columns = get_columns()
    data = get_data(filters)
    return columns, data

def get_data(filters):
    conditions = ''
    
    if filters.get('username'):
        conditions += f" AND tu.username LIKE '%{filters.get('username')}%'"

    if filters.get('role_profile_name'):
        conditions += f" AND tu.role_profile_name = '{filters.get('role_profile_name')}'"

    if filters.get('enabled'):
        enabled_value = frappe.db.escape(filters.get('enabled'))
        if enabled_value == "'Enabled'":
            conditions += " AND tu.enabled = 1"
        elif enabled_value == "'Disabled'":
            conditions += " AND tu.enabled = 0"
    
    query = frappe.db.sql(f"""
                       SELECT tu.email, tu.username, tu.role_profile_name, tu.module_profile, tu.enabled
                       FROM `tabUser` tu
                       WHERE  1=1 {conditions}
                    """)

    return query

def get_columns():
    return [
        "Email:Data:200",
        "Username:Data:200",
        "Role Profile:Link/Role Profile:200",
        "Module Profile:Link/Module Profile:200",
        "Status:Check:150",
    ]